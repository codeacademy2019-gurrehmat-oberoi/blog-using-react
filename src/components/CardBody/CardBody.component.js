import React from 'react';
import './CardBody.css'

const CardBody = (props) => {
  const date = props.date;
  const readTime = props.readTime;
  const blogTitle = props.blogTitle;
  const blogPara = props.blogPara;

  return (
    <div className='CardBody'>
      <div className='meta'>
        <span className="date">{date}</span>
        <span className="readTime">{readTime}</span>
      </div>
      <h1>{blogTitle}</h1>
      <p>{blogPara}</p>
      <hr />
    </div>
  );
}

CardBody.defaultProps = {
  date: '1st January 1970',
  readTime: '1 min',
  blogTitle: 'Lorem ipsum dolor sit',
  blogPara: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
};

export default CardBody;