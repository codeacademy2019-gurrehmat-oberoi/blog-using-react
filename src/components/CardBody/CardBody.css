.CardBody {
  width: 97%;
  padding: 10px;
  color: gray;
  margin: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.CardBody .meta {
  display: flex;
  justify-content: space-between;
}

.CardBody h1 {
  color: black;
}

.CardBody hr {
  /* height: 2px; */
  color: black;
  width: 100%;
}
