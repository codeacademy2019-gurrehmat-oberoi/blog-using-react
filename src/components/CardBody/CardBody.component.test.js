import React from 'react';
import renderer from 'react-test-renderer';
import CardBody from './CardBody.component';

describe('CardBody', () => {
  it('should render a CardBody', () => {
    const tree = renderer.create(<CardBody />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})