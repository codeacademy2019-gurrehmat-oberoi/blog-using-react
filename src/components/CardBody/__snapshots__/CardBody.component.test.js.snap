// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBody should render a CardBody 1`] = `
<div
  className="CardBody"
>
  <div
    className="meta"
  >
    <span
      className="date"
    >
      1st January 1970
    </span>
    <span
      className="readTime"
    >
      1 min
    </span>
  </div>
  <h1>
    Lorem ipsum dolor sit
  </h1>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  </p>
  <hr />
</div>
`;
